src/assets/src/components/preferences.tsx

```diff
@@ -0,0 +1,83 @@
+import * as React from "react";
+import { useState, useEffect } from "react";
+import PhoneInput from "react-phone-input-2";
+import 'react-phone-input-2/lib/style.css'
+
+import * as api from "../services/api";
+import { User } from "../models";
+import { ErrorDisplay, LoadingDisplay, EditToggleField, LoginDialog, Breadcrumbs } from "./common";
+import { usePromise } from "../hooks/usePromise";
+import { useAutoRefresh } from "../hooks/useAutoRefresh";
+import { redirectToLogin } from "../utils";
+import { PageProps } from "./page";
+
+interface PreferencesEditorProps {
+    user: User;
+    disabled: boolean;
+    onUpdateInfo: (phone_number: string) => void;
```

Comment:
All of the attributes in models.ts are snake case; would you want me to change all of them to camel case?

Reply:
Nah, the models should stay how they are - they're reflective of the JSON that is returned by the API, and the API uses snake_case because it's written in Python. (a typescript `interface` just hints to typescript how it expects a data structure to look for type checking purposes)